ocfweb/account/recommender.py

```diff
@@ -0,0 +1,41 @@
+from random import randint
+
+from ocflib.account.creation import validate_username
+
+
+def recommend(real_name, n):
+    name_fields = real_name.split()
+    for i in range(len(name_fields)):
+        name_fields[i] = name_fields[i].lower()
+
+    recs = []
+
+    for i in range(len(name_fields)):
+        rec = ''
+        for j in range(len(name_fields)):
+            if i == j:
+                rec += name_fields[j][0]
+            else:
+                rec += name_fields[j]
```

Comment:
I feel like we can do better than this by systematically generating names in a clever order. I admittedly haven't thought of a way to iterate through _all_ possible names, but we can get a large number of reasonable names using, say, [`itertools.combinations`](https://docs.python.org/3/library/itertools.html#itertools.combinations).

For example, we could try looping over all combinations of names and taking the first `n` letters of those chosen, where `n = 0, 1, 2, ...`:
```python
# Stop at some low n so we don't loop for a million years
for n in range(3):
    for r in range(1, len(name_fields) + 1)
        for c in itertools.combinations(range(len(name_fields)), r):
            try_names = list(name_fields)
            for i in c:
                try_names[i] = try_names[i][:n]
            <test validity of ''.join(try_names)>
```
We'd probably try the full name first of all, though.

Reply:
What do you think about keeping the first letter of each name intact (for "Tony Liu", keep "t" and "l")?
And also how bout only removing vowels (though that might greatly limit the total number of possible recommendations)?